### DISCUSSION TITLE
How create new object JTable?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#33700] [Q&A]
Code in J4.
```
Table::addIncludePath(JPATH_ROOT . '/modules/mod_axitech/tables');
$row = Table::getInstance('EUI');
```
This is working code.
But 
```
Table::addIncludePath(JPATH_ROOT . '/modules/mod_axitech/tables');
$row = new TableEUI;
```
This is NOT working code. This class not exist.
Help me.
How create object from with types?

**Comment by joomdonation** [comment]
If you don't want to use JTable::getInstance , you should use require_once command to require the file which contains your table class and it should work fine.

**Reply by korenevskiy** [reply]
@joomdonation 
I will use `**requare_once**;`
But now there's no point in writing this?
**`Jtable::addIncludePath(JPATH_ROOT . '/modules/mod_axitech/tables');`**

**Reply by joomdonation** [reply]
Sure, in that case `JTable::addIncludePath` is not needed.

**Reply by korenevskiy** [reply]
@joomdonation 
I used to use **`Jtable::getInstance ();`**
but now for the first time I decided to use new Table;
Thanks for the help.
Tell me please.
Does Table have automatic type casting?
I got the object from the **JSON** serialization.
Then I do **`$table->save($json_object,", 'id');`**
there are empty values in the json. But the very type of these fields have integers.
Can Table cast empty field types to their default values?
